cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,305 @@
+type RococoBalances = <RococoRelay as RococoRelayPallet>::Balances;
+type RococoIdentityMigrator = <RococoRelay as RococoRelayPallet>::IdentityMigrator;
+type PeopleRococoIdentity = <PeopleRococo as PeopleRococoPallet>::Identity;
+type PeopleRococoBalances = <PeopleRococo as PeopleRococoPallet>::Balances;
+
+#[derive(Clone, Debug)]
+struct Identity {
+	relay: IdentityInfo<MaxAdditionalFields>,
+	parachain: IdentityInfoParachain,
+}
+
+fn identities() -> Vec<Identity> {
+	let pgp_fingerprint = [
+		0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF0, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+		0x88, 0x99, 0xAA, 0xBB, 0xCC,
+	];
+	let riot = Data::Raw(b"riot-xcm-test".to_vec().try_into().unwrap());
+	let additional = BoundedVec::try_from(vec![(
+		Data::Raw(b"additional".to_vec().try_into().unwrap()),
+		Data::Raw(b"data".to_vec().try_into().unwrap()),
+	)])
+	.unwrap();
+	let first_relay = IdentityInfo {
+		display: Data::Raw(b"xcm-test-one".to_vec().try_into().unwrap()),
+		legal: Data::Raw(b"The Right Ordinal Xcm Test, Esq.".to_vec().try_into().unwrap()),
+		web: Data::Raw(b"https://xcm-test.io".to_vec().try_into().unwrap()),
+		email: Data::Raw(b"xcm-test@gmail.com".to_vec().try_into().unwrap()),
+		pgp_fingerprint: None,
+		image: Data::Raw(b"xcm-test.png".to_vec().try_into().unwrap()),
+		twitter: Data::Raw(b"@xcm-test".to_vec().try_into().unwrap()),
+		riot: Default::default(),
+		additional: Default::default(),
+	};
+	let mut second_relay = first_relay.clone();
+	let mut third_relay = first_relay.clone();
+	let mut fourth_relay = first_relay.clone();
+	second_relay.pgp_fingerprint = Some(pgp_fingerprint);
+	third_relay.pgp_fingerprint = Some(pgp_fingerprint);
+	third_relay.riot = riot.clone();
+	fourth_relay.pgp_fingerprint = Some(pgp_fingerprint);
+	fourth_relay.riot = riot.clone();
+	fourth_relay.additional = additional;
+	let first_parachain = IdentityInfoParachain {
+		display: Data::Raw(b"xcm-test".to_vec().try_into().unwrap()),
+		legal: Data::Raw(b"The Right Ordinal Xcm Test, Esq.".to_vec().try_into().unwrap()),
+		web: Data::Raw(b"https://xcm-test.io".to_vec().try_into().unwrap()),
+		matrix: Data::None,
+		email: Data::Raw(b"xcm-test@gmail.com".to_vec().try_into().unwrap()),
+		pgp_fingerprint: None,
+		image: Data::Raw(b"xcm-test.png".to_vec().try_into().unwrap()),
+		twitter: Data::Raw(b"@xcm-test".to_vec().try_into().unwrap()),
+		github: Data::None,
+		discord: Data::None,
+	};
+	let mut second_parachain = first_parachain.clone();
+	let mut third_parachain = first_parachain.clone();
+	let mut fourth_parachain = first_parachain.clone();
+	second_parachain.pgp_fingerprint = Some(pgp_fingerprint);
+	third_parachain.pgp_fingerprint = Some(pgp_fingerprint);
+	third_parachain.matrix = riot.to_owned(); // riot field on relay is set to matrix in parachain
+	fourth_parachain.pgp_fingerprint = Some(pgp_fingerprint);
+	fourth_parachain.matrix = riot.to_owned();
+	fourth_parachain.github = riot.to_owned();
```

Comment:
This is getting better, but I still think it's not testing the right things. I would make at least the following cases:

1. Test minimal identity.
    - Relay: All set to `Data::None`
    - Para: All set to `Data::None`
2. Test full identity with no `additional`
    - Relay: All set to some non-`None` variant, except `additional`
    - Para: All set to some non-`None` variant, but `github` and `discord` will be `Data::None`
3. Test (2) but with _nonsense_ `additional`
    - Relay: Set `additional` in the Relay, but nothing meaningful. Will be nothing to map it to on the para.
    - Para: Nothing to change w/r/t `additional`
4. Test (2) but with meaningful `additional`
    - Relay: Set `additional` in the Relay, with `[("github", "mygithubname"), ("discord", "mydiscordname")]`
    - Para: Set the `discord` and `github` fields appropriately.

I see no reason to treat Riot/Matrix or PGP keys as special. The field name isn't encoded in the struct and the values are the same.